# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/SharedSurfaceANGLE.cpp
# Commit: edfb73b96d5a
# Full Hash: edfb73b96d5ad946bbdd9da2e98d07b1f4565e1f
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2019-06-15 21:44:24
# Regressor Bug: 1559524
# File Overlap Count: 2
# Description:
#   Bug 1559524 - Reduce calls to GLLibraryEGL::Get(). r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35103
# ==============================================================================

diff -r 0378898329b2 -r edfb73b96d5a gfx/gl/SharedSurfaceANGLE.cpp
--- a/gfx/gl/SharedSurfaceANGLE.cpp	Sat Jun 15 05:14:53 2019 +0000
+++ b/gfx/gl/SharedSurfaceANGLE.cpp	Sat Jun 15 05:21:12 2019 +0000
@@ -42,7 +42,8 @@
 SharedSurface_ANGLEShareHandle::Create(GLContext* gl, EGLConfig config,
                                        const gfx::IntSize& size,
                                        bool hasAlpha) {
-  auto* egl = gl::GLLibraryEGL::Get();
+  const auto& gle = GLContextEGL::Cast(gl);
+  const auto& egl = gle->mEgl;
   MOZ_ASSERT(egl);
   MOZ_ASSERT(egl->IsExtensionSupported(
       GLLibraryEGL::ANGLE_surface_d3d_texture_2d_share_handle));
@@ -306,13 +307,14 @@
     GLContext* gl, const SurfaceCaps& caps,
     const RefPtr<layers::LayersIPCChannel>& allocator,
     const layers::TextureFlags& flags) {
-  auto* egl = gl::GLLibraryEGL::Get();
+  const auto& gle = GLContextEGL::Cast(gl);
+  const auto& egl = gle->mEgl;
   if (!egl) return nullptr;
 
   auto ext = GLLibraryEGL::ANGLE_surface_d3d_texture_2d_share_handle;
   if (!egl->IsExtensionSupported(ext)) return nullptr;
 
-  EGLConfig config = GLContextEGL::Cast(gl)->mConfig;
+  const auto& config = gle->mConfig;
 
   typedef SurfaceFactory_ANGLEShareHandle ptrT;
   UniquePtr<ptrT> ret(new ptrT(gl, caps, allocator, flags, egl, config));